app/src/androidTest/java/com/example/triptracker/e2eTest.kt

```diff
@@ -1,58 +1,141 @@
 import com.example.triptracker.view.profile.UserProfileMyTrips
 import com.example.triptracker.view.profile.UserProfileOverview
 import com.example.triptracker.view.profile.UserProfileSettings
+import com.example.triptracker.viewmodel.HomeViewModel
+import com.example.triptracker.viewmodel.MapViewModel
+import com.example.triptracker.viewmodel.UserProfileViewModel
+import io.github.kakaocup.compose.node.element.ComposeScreen
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.junit4.MockKRule
+import io.mockk.mockk
+import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class E2ETest {
+  private val appContext: Context = InstrumentationRegistry.getInstrumentation().targetContext
 
   @get:Rule val composeTestRule = createComposeRule()
 
+  // This rule automatic initializes lateinit properties with @MockK, @RelaxedMockK, etc.
+  @get:Rule val mockkRule = MockKRule(this)
+
+  // Relaxed mocks methods have a default implementation returning values
+
+  @RelaxedMockK lateinit var mockNav: Navigation
+  @RelaxedMockK private lateinit var homeViewModel: HomeViewModel
+  @RelaxedMockK private lateinit var mapViewModel: MapViewModel
+
+  @RelaxedMockK private lateinit var mockItineraryRepository: ItineraryRepository
+  @RelaxedMockK private lateinit var mockUserProfileRepository: UserProfileRepository
+  @RelaxedMockK private lateinit var mockUserProfileViewModel: UserProfileViewModel
+  @RelaxedMockK private lateinit var mockProfile: MutableUserProfile
   var profile = MutableUserProfile()
+  val mockList = MockItineraryList()
+  val mockItineraries = mockList.getItineraries()
+
+  val mockUserList = MockUserList()
+  val mockUsers = mockUserList.getUserProfiles()
+  val mockMail = "test@gmail.com"
+
+  @Before
+  fun setUp() {
+    // Mocking necessary components
+    mockNav = mockk(relaxed = true)
+    mockItineraryRepository = mockk(relaxed = true)
+    homeViewModel = mockk(relaxed = true)
+    mapViewModel = mockk(relaxed = true)
+    // This allows mocking unit returning functions
+    mockUserProfileRepository = mockk(relaxUnitFun = true)
+    mockUserProfileViewModel = mockk(relaxUnitFun = true)
+
+    MockKAnnotations.init(this, relaxUnitFun = true)
+    mockProfile = mockk(relaxUnitFun = true)
+
+    // Log.d("ItineraryList", mockViewModel.itineraryList.value.toString())
+    every { mockNav.getTopLevelDestinations()[0] } returns
+        TopLevelDestination(Route.HOME, Icons.Outlined.Home, "Home")
+
+    every { mockUserProfileViewModel.fetchAllUserProfiles { any() } } answers
+        {
+          // Invoke the callback with mock data
+          val callback = arg<(List<UserProfile>) -> Unit>(0)
+          callback(mockUsers)
+        }
+  }
 
-  /*
-     Test to check the flow of the profile screen navigating through different screens
-  */
```

Comment:
Why did you remove the comment? Should we add a different comment instead?

Reply:
Because its not about the profile screen anymore but a global end to end !